src/ShellClients/HideTracker.vala

```diff
@@ -6,14 +6,30 @@
  */
 
 public class Gala.HideTracker : Object {
-    private const uint UPDATE_TIMEOUT = 200;
+    private const int BARRIER_OFFSET = 50; // Allow hot corner trigger
+    private const int UPDATE_TIMEOUT = 200;
+    private const int HIDE_DELAY = 500;
 
     public signal void hide ();
     public signal void show ();
 
     public Meta.Display display { get; construct; }
     public unowned PanelWindow panel { get; construct; }
-    public Pantheon.Desktop.HideMode hide_mode { get; set; default = NEVER; }
+
+    private Pantheon.Desktop.HideMode _hide_mode = NEVER;
+    public Pantheon.Desktop.HideMode hide_mode {
+        get {
+            return _hide_mode;
+        }
+        set {
+            _hide_mode = value;
+
+            destroy_barrier ();
+            if (hide_mode != NEVER) {
```

Comment:
HideTracker will actually never exist with a NEVER hide mode. So maybe we can just remove all this here as well as the destroy_barrier method and connect setup_barrier to a notify["hide-mode"]? Would remove a lot of code and make it IMHO cleaner.

Reply:
I prefer using setters instead of notify in cases where we connect notify to itself. Using notify just moves the code to the constructor but it doesn't make it any cleaner IMO.

Reply:
Well we would go from

```vala
    private Pantheon.Desktop.HideMode _hide_mode = NEVER;
    public Pantheon.Desktop.HideMode hide_mode {
        get {
            return _hide_mode;
        }
        set {
            _hide_mode = value;

            setup_barrier ();
        }
    }
```
to
```vala
public Pantheon.Desktop.HideMode hide_mode { get; set; }
…
notify["hide-mode"].connect (setup_barrier);
```
which is in the very least easier to read because you don't have to skip a lot of boilerplate code everytime you skim over the class IMHO :shrug:

Reply:
The biggest issue I see in this approach is `...`. The property definition and what happens when you change it will be from 50 to 100 lines apart. For me that's counter intuitive and hard to read.

Reply:
While this definitely makes sense and it isn't a no go for me I still prefer the notify (which is also used e.g. in PanelClone - admittedly also written by me but consistency etc. :shrug: ) so maybe we should just a let a third party decide? :)

Reply:
I understand the urge to remove the boilerplate code. However for me, the setter/getter approach is more readable right away. The notify approach is new for me and I need to read about it.

Reply:
Another vote for the setter approach, if the property name ever changes there’s no compile time errors with notify so the setter way is safer imo

Reply:
Then this is good to go from my side :)